### ISSUE TITLE
Custom SSL prefix

### ISSUE THREAD

**Issue by DomT4** [vanhauser-thc/thc-hydra#80]
Something of a feature request - The ability to pass a custom prefix to the SSL libraries/headers.

As you may well be aware Apple removed the OpenSSL headers from OS X 10.11 and Xcode 7 (Which impacts OS X 10.10 as well now) and now `hydra` refuses to compile on those platforms with SSL support *unless* you have an SSL installation in another default-searched path as defined by `configure`. 

It's not always practical to put things in specific search paths so was wondering if you'd consider adding support for a custom SSL prefix?

I tried to throw something together in line with your current usage but my patience with `configure` scripts and autotools in general is not the highest and it doesn't quite work.

```diff
diff --git a/configure b/configure
index 8980e24..ef6e0ab 100755
--- a/configure
+++ b/configure
@@ -8,6 +8,8 @@ if [ "$1" = "-h" -o "$1" = "--help" ]; then
   echo "  --prefix=path              path to install hydra and its datafiles to"
   echo "  --with-oracle=prefix       prefix for oracle include dir"
   echo "  --with-oracle-lib=prefix   prefix for oracle lib dir"
+  echo "  --with-ssl=prefix          prefix for SSL headers"
+  echo "  --with-ssl-lib=prefix      prefix for SSL libraries"
   echo "  --disable-xhydra           disable compilation of hydra GUI"
   echo "  --nostrip                  do not per default strip binaries before install"
   echo "  --debug                    show debug output to trace errors"
@@ -25,6 +27,8 @@ WORACLE_PATH=""
 WORACLE_LIB_PATH=""
 SSL_PATH=""
 SSL_IPATH=""
+WSSL_PATH=""
+WSSL_LIB_PATH=""
 CURSES_PATH=""
 CURSES_IPATH=""
 CRYPTO_PATH=""
@@ -145,10 +149,14 @@ if [ "X" != "X$DEBUG" ]; then
    echo DEBUG: SSL_INC=$INCDIRS `ls -d /*ssl/include /opt/*ssl/include /usr/*ssl/include /usr/local/*ssl/include 2> /dev/null`
 fi
 
-for i in $LIBDIRS \
-/*ssl /usr/*ssl /opt/*ssl /usr/local/*ssl /opt/local/*ssl \
-/*ssl/lib /usr/*ssl/lib /opt/*ssl/lib /usr/local/*ssl/lib /opt/local/*ssl/lib
-do
+if [ "X" = "X$WSSL_LIB_PATH" ]; then
+    SSL_PATH="$i"
+    CRYPTO_PATH="$i"
+else
+  for i in $LIBDIRS \
+  /*ssl /usr/*ssl /opt/*ssl /usr/local/*ssl /opt/local/*ssl \
+  /*ssl/lib /usr/*ssl/lib /opt/*ssl/lib /usr/local/*ssl/lib /opt/local/*ssl/lib
+  do
     if [ "X" = "X$SSL_PATH" ]; then
         if [ -f "$i/libssl.so" -o -f "$i/libssl.dylib" -o -f "$i/libssl.a" ]; then
             SSL_PATH="$i"
@@ -171,18 +179,24 @@ do
           CRYPTO_PATH="$i"
         fi
     fi
-done
+  done
+fi
 
 SSLNEW=""
-for i in $INCDIRS /*ssl/include /opt/*ssl/include /usr/*ssl/include /usr/local/*ssl/include
-do
+if [ "X" = "X$WSSL_PATH" ]; then
+      SSL_PATH="$i"
+      SSLNEW=`grep SHA256_CTX $i/openssl/sha.h 2> /dev/null`
+else
+  for i in $INCDIRS /*ssl/include /opt/*ssl/include /usr/*ssl/include /usr/local/*ssl/include
+  do
     if [ "X" = "X$SSL_IPATH" ]; then
         if [ -f "$i/openssl/ssl.h" ]; then
             SSL_IPATH="$i"
             SSLNEW=`grep SHA256_CTX $i/openssl/sha.h 2> /dev/null`
         fi
     fi
-done
+  done
+fi
 if [ "X" != "X$DEBUG" ]; then
    echo DEBUG: SSL_PATH=$SSL_PATH/libssl
    echo DEBUG: CRYPTO_PATH=$CRYPTO_PATH/libcrypto
```

**Comment by vanhauser-thc** [comment]
thanks, looks useful.
I imported the patch, please check if everything is fine.

(next time you can push your patch upstream via github :) )

**Comment by DomT4** [comment]
I couldn't quite get the patch working yesterday which is why I left it in an issue :). It doesn't break the build at all, but it doesn't work in pointing at the directory yet. I'll keep working on it and submit an update via PR when I get it to function as intended locally.

**Comment by Hamid-K** [comment]
The new configure script committed into the repository is broken, and regardless of availablity of libssl / libcrypto always returns fail, thus not allowing compilation with SSL support. reverting back to the previous revision of configure file temporarily solved my problem.

**Comment by DomT4** [comment]
Yes, the aim wasn't to get it committed just yet. Issue was more a question about the current logic in the script. The `diff` was more or less a *"This is what I'm working on & why I'm asking"* :smile_cat:. I couldn't actually break it though, locally, even without the new logic working as intended yet.

What OS are you using?

**Comment by Hamid-K** [comment]
yes I understand it's not meant to be  a final working revision based on your initial comment. 
ubuntu 14 64. Even manually specifying lib path via parameters won't make it work. libs are installed in standard path.

**Comment by SecurityAddicted** [comment]
I'm having the same problem... MacOSX 10.11 with OpenSSL installed through Macports.
The script can't find OpenSSL... tried to specify paths but still nothing.

**Comment by dirkjanm** [comment]
I don't know if the fix in #84 still allows the custom prefix, but at least it compiled again on Ubuntu and Kali without specifying a prefix.
